kconfig-hardened-check.py

```diff
@@ -452,31 +465,34 @@ def check_config_file(fname):
                         help='check the config_file against these preferences')
     parser.add_argument('--debug', action='store_true',
                         help='enable internal debug mode')
+    parser.add_argument('--json', action='store_true',
+                        help='output results in json format')
     args = parser.parse_args()
 
     if args.debug:
         debug_mode = True
 
     if args.config:
-        arch, msg = detect_arch(args.config)
+        arch, msg = detect_arch(args.json, args.config)
```

Comment:
Let's create some global variable to avoid passing `args.json` to `detect_arch()` and `check_config_file()` - it doen't really matter for their main logic.
It will be similar to `debug_mode`.